--- XML-Twig-3.29/Makefile.PL.orig	2007-03-26 18:09:58.000000000 +0200
+++ XML-Twig-3.29/Makefile.PL	2007-03-26 18:13:29.000000000 +0200
@@ -22,7 +22,7 @@
 else
   { foreach my $prompt (@prompts)
       { my ($program, $default, $description) = @$prompt;
-        if( prompt("Do you want to install '$program' ($description)?", $default) =~ /^y/)
+#        if( prompt("Do you want to install '$program' ($description)?", $default) =~ /^y/)
           { push(@programs, $program); }
       }
   }
