--- initng-0.1.8/tools/install_service.c.orig	2005-08-17 19:24:25.000000000 +0200
+++ initng-0.1.8/tools/install_service.c	2005-08-17 19:29:46.000000000 +0200
@@ -64,6 +64,8 @@
         return ("gentoo");
     if (is_file("/etc/fedora-release"))
         return ("fedora");
+    if (is_file("/etc/slackware-version"))
+        return ("slackware");
     D_("Probe failed\n");
     return ("unknown");
 }
