From: Robert O'Callahan <rocallahan@novell.com>
Subject: fixed crash in system-prefs
References:
https://bugzilla.mozilla.org/show_bug.cgi?id=313962

Index: mozilla/extensions/pref/system-pref/src/nsSystemPref.cpp
===================================================================
RCS file: /home/rocallahan/mozilla-cvs-mirror/mozilla/extensions/pref/system-pref/src/nsSystemPref.cpp,v
retrieving revision 1.9
diff -u -t -p -1 -2 -r1.9 nsSystemPref.cpp
--- mozilla/extensions/pref/system-pref/src/nsSystemPref.cpp	30 Apr 2005 02:37:41 -0000	1.9
+++ mozilla/extensions/pref/system-pref/src/nsSystemPref.cpp	27 Oct 2005 01:13:16 -0000
@@ -51,24 +51,26 @@ union MozPrefValue {
     char *      stringVal;
     PRInt32     intVal;
     PRBool      boolVal;
 };
 
 struct SysPrefItem {
     const char *prefName;       // mozilla pref string name
     MozPrefValue defaultValue;  // store the mozilla default value
     PRBool isLocked;  // store the mozilla lock status
     SysPrefItem() {
         prefName = nsnull;
         defaultValue.intVal = 0;
+        defaultValue.stringVal = nsnull;
+        defaultValue.boolVal = PR_FALSE;
         isLocked = PR_FALSE;
     }
     void SetPrefName(const char *aPrefName) {
         prefName = aPrefName;
     }
 };
 
 // all prefs that mozilla need to read from host system if they are available
 static const char *sSysPrefList[] = {
     "network.proxy.http",
     "network.proxy.http_port",
     "network.proxy.ftp",
