--- aterm-0.4.2~/src/feature.h	Thu Sep  6 20:38:07 2001
+++ aterm-0.4.2/src/feature.h	Sun Jan  6 03:29:31 2002
@@ -175,7 +175,7 @@
  * Default separating chars for multiple-click selection
  * Space and tab are separate separating characters and are not settable
  */
-#define CUTCHARS	"\"&'()*,;<=>?@[\\]^`{|}~"
+#define CUTCHARS	"\"&'()*,;<=>?@[\\]^`{|}:"
 
 /*
  * Add run-time support for changing the cutchars for double click selection
