--- gtkmm-2.4.8/Makefile.in.orig	2005-01-10 17:33:08.000000000 +0100
+++ gtkmm-2.4.8/Makefile.in	2005-01-10 17:33:42.000000000 +0100
@@ -217,7 +217,7 @@
 web_path_gtkmm2 = /home/groups/g/gt/gtkmm/htdocs/docs/gtkmm-2.4/
 web_path_docs = $(web_path_gtkmm2)docs/
 rsync_args = -vz --rsh ssh --delete --delete-after
-gtkmm_docdir = $(datadir)/doc/gtkmm-2.4/docs
+gtkmm_docdir = $(prefix)/doc/gtkmm-2.4/docs
 doc_tarball_files = \
 	docs/index.html docs/FAQ/html docs/images/*.gif \
 	docs/internal/*.txt docs/internal/*.dia docs/reference/html \
