--- redhat-artwork-0.117/art/gdm/Bluecurve/Bluecurve.xml.in.orig	2004-12-19 01:56:09.000000000 +0100
+++ redhat-artwork-0.117/art/gdm/Bluecurve/Bluecurve.xml.in	2004-12-19 01:56:28.000000000 +0100
@@ -191,10 +191,10 @@
   <item type="rect">
     <pos anchor="c" x="50%" y="45%" width="100%" height="box"/>
     <box orientation="vertical" min-width="300" spacing="10">
-      <item type="pixmap">
+<!--      <item type="pixmap">
         <pos anchor="n" x="50%"/>
         <normal file="rh_logo-header.png"/>
-      </item>
+      </item> -->
       <item type="rect">
         <pos anchor="n" x="50%" width="100%" height="box"/>
 	<box orientation="horizontal" spacing="10" homogeneous="true">
