--- k3b-0.11.23.orig/src/device/k3bscsicommand.h	2004-01-21 11:20:11.000000000 +0100
+++ k3b-0.11.23/src/device/k3bscsicommand.h	2005-03-27 13:29:54.000000000 +0200
@@ -16,6 +16,12 @@
 #ifndef _K3B_SCSI_COMMAND_H_
 #define _K3B_SCSI_COMMAND_H_
 
+#include >linux/version.h>
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,4,21)
+typedef unsigned long long__u64;
+#endif
+
 #include <sys/types.h>
 #undef __STRICT_ANSI__
 #include <linux/cdrom.h>
