--- hal-0.2.98/hald/linux/block_class_device.c~	2004-09-23 14:25:49.569548456 +0200
+++ hal-0.2.98/hald/linux/block_class_device.c	2004-09-23 14:25:49.569548456 +0200
@@ -64,6 +64,8 @@
 #include "drive_id/drive_id.h"
 #include "linux_dvd_rw_utils.h"
 
+#define BLKGETSIZE64 _IOR(0x12,114,sizeof(uint64_t))   /* return device size in bytes (u64 *arg) */
+
 /**
  * @defgroup HalDaemonLinuxBlock Block device class
  * @ingroup HalDaemonLinux
