? f.diff
Index: SlackBuild
===================================================================
RCS file: /cvsroot/portpkg/ports/ap/streamripper/SlackBuild,v
retrieving revision 1.2
diff -u -b -B -r1.2 SlackBuild
--- SlackBuild	19 Nov 2007 15:53:04 -0000	1.2
+++ SlackBuild	13 Dec 2007 10:44:48 -0000
@@ -1,8 +1,7 @@
 #!/bin/sh
 # MAINTAINER: Vadim Shevchenko 'shv' <shevadim@mail.ru>
-# SOURCES: http://belnet.dl.sourceforge.net/sourceforge/streamripper/streamripper-1.62.1.tar.gz
-# MD5SUMS: d92a37bf23ee3f1bd5a9498d72c0562a streamripper-1.61.27.tar.gz
-# MD5SUMS: c8c84b8acf84f3d431bf7b6303acc63b streamripper-1.62.1.tar.gz
+# SOURCES: http://belnet.dl.sourceforge.net/sourceforge/streamripper/streamripper-1.62.3.tar.gz
+# MD5SUMS: 127eb52355d0d10183ee22c44b5a9039 streamripper-1.62.3.tar.gz
 # REQUIRES: libmad libogg libvorbis
 # OPTIONAL: faad2
 
@@ -11,7 +10,7 @@
 NAME=`basename $CWD`
 PKG=$TMP/package-$NAME
 
-VERSION=1.62.1
+VERSION=1.62.3
 ARCH=${ARCH:-i486}
 BUILD=1shv
 
Index: footprint
===================================================================
RCS file: /cvsroot/portpkg/ports/ap/streamripper/footprint,v
retrieving revision 1.2
diff -u -b -B -r1.2 footprint
--- footprint	19 Nov 2007 15:53:04 -0000	1.2
+++ footprint	13 Dec 2007 10:44:48 -0000
@@ -1,3 +1,8 @@
+PACKAGE SIZE (compressed):      172 K
+PACKAGE SIZE (uncompressed):    390 K
+REMOTE SOURCES:  1229 K
+http://belnet.dl.sourceforge.net/sourceforge/streamripper/streamripper-1.62.3.tar.gz
+FILE LIST:
 drwxr-xr-x  root/root  ./
 drwxr-xr-x  root/root  install/
 -rw-r--r--  root/root  install/doinst.sh
@@ -5,15 +10,14 @@
 drwxr-xr-x  root/root  usr/
 drwxr-xr-x  root/root  usr/bin/
 -rwxr-xr-x  root/root  usr/bin/agrep
--rwxr-xr-x  root/root  usr/bin/cstreamripper
 -rwxr-xr-x  root/root  usr/bin/streamripper
 drwxr-xr-x  root/root  usr/doc/
-drwxr-xr-x  root/root  usr/doc/streamripper-1.62.1/
--rw-r--r--  root/root  usr/doc/streamripper-1.62.1/CHANGES
--rw-r--r--  root/root  usr/doc/streamripper-1.62.1/COPYING
--rw-r--r--  root/root  usr/doc/streamripper-1.62.1/INSTALL
--rw-r--r--  root/root  usr/doc/streamripper-1.62.1/README
--rw-r--r--  root/root  usr/doc/streamripper-1.62.1/THANKS
+drwxr-xr-x  root/root  usr/doc/streamripper-1.62.3/
+-rw-r--r--  root/root  usr/doc/streamripper-1.62.3/CHANGES
+-rw-r--r--  root/root  usr/doc/streamripper-1.62.3/COPYING
+-rw-r--r--  root/root  usr/doc/streamripper-1.62.3/INSTALL
+-rw-r--r--  root/root  usr/doc/streamripper-1.62.3/README
+-rw-r--r--  root/root  usr/doc/streamripper-1.62.3/THANKS
 drwxr-xr-x  root/root  usr/include/
 drwxr-xr-x  root/root  usr/include/tre/
 -rw-r--r--  root/root  usr/include/tre/regex.h
