--- hibernate-script-1.05/hibernate.conf.orig	2005-02-21 14:35:56.000000000 +0100
+++ hibernate-script-1.05/hibernate.conf	2005-02-21 14:35:29.000000000 +0100
@@ -63,7 +63,7 @@
 # AlwaysForce yes
 # AlwaysKill yes
 # SwsuspVT 15
-# Distribution debian (not required - autodetection should work)
+Distribution slackware
 # XDisplay :0
 
 ##############################################################################
