--- hibernate-script-1.05/scriptlets.d/services.orig	2005-02-21 14:34:47.000000000 +0100
+++ hibernate-script-1.05/scriptlets.d/services	2005-02-21 14:35:07.000000000 +0100
@@ -75,9 +75,12 @@
 	suse|mandrake)
 	    INITDIR=/etc/init.d
 	    ;;
-	debian|slackware|redhat|fedora)
+	debian|redhat|fedora)
 	    INITDIR=/etc/init.d
 	    ;;
+	slackware)
+	    INITDIR=/etc/rc.d
+	    ;;
     	*)
 	    # Auto-detect
 	    if [ -d "/etc/init.d/" ] ; then
