--- wifi_radar-1.1/wifi_radar.py.old	2005-02-21 23:22:12.000000000 +0100
+++ wifi_radar-1.1/wifi_radar.py	2005-02-21 23:23:35.000000000 +0100
@@ -7,16 +7,16 @@
 import time, os, sys, re, ConfigParser
 ####################################################################################################
 # Defaults, these could be different for your distro.
-CONF_FILE		= "/etc/conf.d/wifi_radar.conf"
-IWLIST_COMMAND	= "/usr/sbin/iwlist"
-IWCONFIG_COMMAND= "/usr/sbin/iwconfig"
+CONF_FILE		= "/etc/wifi_radar.conf"
+IWLIST_COMMAND	= "/sbin/iwlist"
+IWCONFIG_COMMAND= "/sbin/iwconfig"
 IFCONFIG_COMMAND= "/sbin/ifconfig"
 DHCP_COMMAND	= "/sbin/dhcpcd"
 ROUTE_COMMAND	= "/sbin/route"
-SAY_COMMAND		= "/usr/local/bin/say"
+SAY_COMMAND		= "/usr/bin/say"
 # More defaults, these may get overridden by values found in the conf file
 ICON_FILE		= "/usr/share/pixmaps/wifi_radar.svg"
-INTERFACE		= "eth1"
+INTERFACE		= "eth0"
 SCAN_TIMEOUT	= "5"
 SPEAK_UP		= True
 ####################################################################################################
